distgen/distro_version.py

```diff
@@ -0,0 +1,22 @@
+import platform
+import subprocess
+
+
+def detect_default_distro():
+    os, version, name = platform.dist()
+    os = os.lower()
+
+    if os == 'fedora':
+        if name.lower() == 'rawhide':
+            version = 'rawhide'
+    elif os in ['centos', 'rhel']:
+        pass
+    else:
+        return None
+
+    # Only RPM distros for now:
+    cmd = ['rpm', '--eval', '%_arch']
+    sp = subprocess.Popen(cmd, stdout=subprocess.PIPE)
```

Comment:
there is a possible race here: you should explicitly wait for the command to finish, I encourage to use wrapper `subprocess.check_output` for this

Reply:
Can you elaborate on the race?  I'm just curious..

Nvmd though, check_output is really even semantically better.

Reply:
If the subprocess didn't finish before you tried to read its output, you'd read empty string (or partial result). `check_output` explicitly waits for the subprocess to finish and then returns whole output.

Reply:
But the docs [1] claim that read() without argument should read the entire file.
[1] https://docs.python.org/3/tutorial/inputoutput.html#methods-of-file-objects

Reply:
Ok, I can't access the old version of the code any more and was assuming you were trying to do something else. I'm not sure then, maybe Tomas will explain.

Reply:
It's actually instance of https://docs.python.org/3/library/io.html#io.BufferedReader